### PR TITLE
error out when credentials are empty

### PR DESCRIPTION
**Pull request by gajicdev**
Related to https://gitlab.nine.ch/gcp/mono-infra/-/issues/769

### CODE REVIEW

File: api/util/apps.go [hunk 1]
```diff
@@ -117,6 +117,22 @@ func (git GitAuth) Enabled() bool {
 	return false
 }
 
+func (git GitAuth) Valid() bool {
```
Comment (thirdeyenick) [inline api/util/apps.go:120]: I think we could actually return either a `(bool,error)` or just an `error` to indicate which part has been empty?

Comment (gajicdev) [reply api/util/apps.go]: changed it to an error 👍

File: api/util/apps.go [hunk 2]
```diff
@@ -117,6 +117,22 @@ func (git GitAuth) Enabled() bool {
...
 
+func (git GitAuth) Valid() error {
+	if git.SSHPrivateKey != nil {
+		if *git.SSHPrivateKey == "" {
+			return fmt.Errorf("the SSH private key cannot be empty")
+		}
+	}
+
+	if git.Username != nil && git.Password != nil {
+		if *git.Username == "" && *git.Password == "" {
```
Comment (thirdeyenick) [inline api/util/apps.go:128]: I think a logical "or" might be better here as both, the username or the password, should not be empty?

Comment (gajicdev) [reply api/util/apps.go]: yup you are right, added it

**Review by thirdeyenick — APPROVED** [review]
👍

**Pull request merged**